# Workaround to provide RX buffers that are large enough to hold most frames that Firecracker will
# send.
#
# References:
#   - https://github.com/firecracker-microvm/firecracker/issues/3905
#   - https://docs.oasis-open.org/virtio/virtio/v1.1/cs01/virtio-v1.1-cs01.html#x1-2050006 (section 5.1.6.3)

--- sys/dev/virtio/network/if_vtnet.c
+++ sys/dev/virtio/network/if_vtnet.c
@@ -1240,8 +1240,7 @@ vtnet_rx_cluster_size(struct vtnet_softc *sc, int mtu)
 	else if (framesz <= MJUM9BYTES)
 		return (MJUM9BYTES);
 
-	/* Sane default; avoid 16KB clusters. */
-	return (MCLBYTES);
+	return (MJUM16BYTES);
 }
 
 static int
